packages/i18n/src/messages/localizedConstants.tsx

```diff
@@ -1943,14 +1916,53 @@ const behaviouralSkillLevelDefinitions = defineMessages({
   },
 });
 
-export const getBehaviouralSkillLevelDefinition = (
-  skillLevel: string | number,
-): MessageDescriptor =>
-  getOrThrowError(
+export const getSkillLevelName = (
+  skillLevel: string | SkillLevel,
+  skillCategory: SkillCategory,
+): MessageDescriptor => {
+  if (skillCategory === SkillCategory.Technical) {
+    return getOrThrowError(
+      technicalSkillLevelNames,
+      skillLevel,
+      `Invalid technical skill level '${skillLevel}'`,
+    );
+  }
+
+  return getOrThrowError(
+    behaviouralSkillLevelNames,
+    skillLevel,
+    `Invalid behavioural skill level '${skillLevel}'`,
+  );
+};
+
+export const getSkillLevelDefinition = (
+  skillLevel: string | SkillLevel,
+  skillCategory: SkillCategory,
+): MessageDescriptor => {
+  if (skillCategory === SkillCategory.Technical) {
+    getOrThrowError(
```

Comment:
We will need a return here or we will always get back behavioral ones.

```suggestion
    return getOrThrowError(
```